File: python/customization_single_word.py
from soniox.transcribe_file import transcribe_file_short
from soniox.speech_service import SpeechClient
from soniox.speech_service import SpeechContext, SpeechContextEntry


def main():
    with SpeechClient() as client:
        # Create SpeechContext.
        speech_context = SpeechContext(
            entries=[
                SpeechContextEntry(
                    phrases=["acetylcarnitine"],
                    boost=20,
                ),
                SpeechContextEntry(
                    phrases=["zestoretic"],
                    boost=20,
                ),
            ]
        )

        # Pass SpeechContext with transcribe request.
        result = transcribe_file_short(
            "../test_data/acetylcarnitine_zestoretic.flac",
            client,
            speech_context=speech_context,
        )

        print(" ".join(word.text for word in result.words))


if __name__ == "__main__":
    main()

File: node/customization_single_word.js
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  // Create SpeechContext.
  const speech_context = {
    entries: [
      {
        phrases: ["acetylcarnitine"],
        boost: 15,
      },
      {
        phrases: ["Zestoretic"],
        boost: 15,
      }
    ]
  };

  // Pass SpeechContext with transcribe request.
  const result = await speechClient.transcribeFileShort(
    "../test_data/acetylcarnitine_zestoretic.flac",
    {
      model: "en_v2",
      speech_context: speech_context,
    }
  );

  console.log(result.words.map((word) => word.text).join(""));
})();
